### PR TITLE
Add support for user delegation key

### PR DESCRIPTION
**Pull request by EmmaZhu**
Thanks for contribution! Please go through following checklist before sending PR.

### PR Branch Destination

- For Azurite V3, please send PR to `main` branch.
- For legacy Azurite V2, please send PR to `legacy-dev` branch.

### Always Add Test Cases

Make sure test cases are added to cover the code change.

### Add Change Log

Add change log for the code change in `Upcoming Release` section in `ChangeLog.md`.

### Development Guideline

Please go to CONTRIBUTION.md for steps about setting up development environment and recommended Visual Studio Code extensions.

### CODE REVIEW

File: src/blob/generated/utils/utils.ts [hunk 1]
```diff
@@ -18,3 +20,23 @@ export function isURITemplateMatch(url: string, template: string): boolean {
   }
   return true;
 }
+
+export function getUserDelegationKeyValue(
```
Comment (XiaoningLiu) [inline src/blob/generated/utils/utils.ts:24]: Files under `generated` folder are supposed to be auto generated cross all services.
Should move the UDK utilities somewhere else.

File: src/blob/handlers/ServiceHandler.ts [hunk 2]
```diff
@@ -77,12 +79,31 @@ export default class ServiceHandler extends BaseHandler
...
 
-  public getUserDelegationKey(
+  public async getUserDelegationKey(
     keyInfo: Models.KeyInfo,
     options: Models.ServiceGetUserDelegationKeyOptionalParams,
     context: Context
   ): Promise<Models.ServiceGetUserDelegationKeyResponse> {
-    throw new NotImplementedError(context.contextId);
+    const keyValue = getUserDelegationKeyValue(
+      AAD_OBJECT_ID,
```
Comment (XiaoningLiu) [inline src/blob/handlers/ServiceHandler.ts:88]: object id and tenant id should be extracted from incoming bearer token.

**Review by XiaoningLiu — CHANGES_REQUESTED** [review]
Added comments inline

File: src/blob/generated/utils/utils.ts (hunk 1 above)
Comment (EmmaZhu) [reply src/blob/generated/utils/utils.ts]: Updated.

File: src/blob/handlers/ServiceHandler.ts (hunk 2 above)
Comment (EmmaZhu) [reply src/blob/handlers/ServiceHandler.ts]: Updated.

**Review by XiaoningLiu — APPROVED** [review]

**Review by blueww — APPROVED** [review]

**Pull request merged**